curriculum/challenges/english/25-front-end-development/lab-one-time-password-generator/67c562286b29447da020d407.md

```diff
@@ -0,0 +1,346 @@
+button.click();
+setTimeout(() => {
+  const p = document.querySelector(".container p#otp-timer");
+  assert.match(p.textContent, /Expires in: [1-5] seconds/);
+}, 2000);
+```
+
+The `"Generate OTP"` button should be disabled while the countdown timer is running.
+
+```js
+const button = document.querySelector(".container button#generate-otp-button");
+assert.exists(button, "The Generate OTP button should exist");
+
+button.dispatchEvent(new Event('click'));
+
+setTimeout(() => {
+  assert.isTrue(button.disabled);
+}, 2000);
+
+setTimeout(() => {
+  assert.isFalse(button.disabled);
+}, 6000);
+```
+
+When the countdown timer reaches `0`, you should display the message `OTP expired. Click the button to generate a new OTP.`.
+
+```js
+const button = document.querySelector(".container button#generate-otp-button");
+button.dispatchEvent(new Event('click'));
+
+setTimeout(() => {
+  const p = document.querySelector(".container p#otp-timer");
+  assert.equal(p.textContent, "OTP expired. Click the button to generate a new OTP.");
+}, 6000);
+```
+
+The generated OTP should be 6 digits long.
+
+```js
+const button = document.querySelector(".container button#generate-otp-button");
+const h2 = document.querySelector(".container h2#otp-display");
+button.dispatchEvent(new Event('click'));
+
+assert.match(h2.textContent, /^[0-9]{6}$/, "OTP should be a 6-digit number");
+```
+
+You should export the `OTPGenerator` component.
+
+```js
+assert.isFunction(window.index.OTPGenerator, 1);
+```
+
+# --seed--
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+    <meta charset="UTF-8" />
+    <title>Color Picker</title>
+    <link rel="stylesheet" href="styles.css" />
+    <script src="https://unpkg.com/react@18/umd/react.development.js"></script>
+    <script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"></script>
+    <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
+    <script
+      data-plugins="transform-modules-umd"
+      type="text/babel"
+      src="index.jsx"
+    ></script>
+</head>
+
+<body>
+    <div id="root"></div>
+    <script
+      data-plugins="transform-modules-umd"
+      type="text/babel"
+      data-presets="react"
+      data-type="module"
+    >
+      import { OTPGenerator } from './index.jsx';
+      ReactDOM.createRoot(document.getElementById('root')).render(<OTPGenerator />);
+    </script>
+</body>
+
+</html>
+```
+
+```css
+
+```
+
+```jsx
+const { useState, useEffect, useRef } = React;
+
+export const OTPGenerator = () => {};
+```
+
+# --solutions--
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8" />
+    <title>OTP Generator</title>
+    <script src="https://unpkg.com/react@18/umd/react.development.js"></script>
+    <script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js"></script>
+    <script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>
```

Comment:
```suggestion
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react/18.3.1/umd/react.development.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/18.3.1/umd/react-dom.development.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/babel-standalone/7.26.5/babel.min.js"></script>
```
We're using cdnjs since unpkg suffered a few outages: https://github.com/freeCodeCamp/freeCodeCamp/pull/59291